packages/eds-core-react/src/components/SideBar/SideBarAccordion/index.tsx

```diff
@@ -0,0 +1,320 @@
+  const { isOpen } = useSideBar()
+
+  const combinedRefs = useMemo(
+    () => mergeRefs<HTMLButtonElement>(setAnchorEl, ref),
+    [anchorEl, ref],
+  )
+
+  const closeMenu = () => {
+    setMenuIsOpen(false)
+  }
+
+  const onClickWhenSidePanelExpanded = () => {
+    if (toggleExpand === undefined) {
+      setAccordionIsOpen(!accordionIsOpen)
+    } else {
+      toggleExpand()
+    }
+    onClick && onClick()
+  }
+
+  const onClickWhenSidePanelClosed = () => {
+    setMenuIsOpen(!menuIsOpen)
+    onClick && onClick()
+  }
+
+  const getIconColor = () => {
+    return active ? typographyActiveColor : typographyColor
+  }
+
+  const showPanel = toggleExpand !== undefined ? isExpanded : accordionIsOpen
+
+  if (isOpen) {
+    return (
+      <MenuItem $active={active} $isExpanded={showPanel}>
+        <AccordionHeader onClick={onClickWhenSidePanelExpanded}>
+          <AccordionButton
+            id={`header_${accordionId}`}
+            aria-expanded={showPanel}
+            aria-controls={`panel_${accordionId}`}
+            ref={ref}
+            {...rest}
+          >
+            {icon && <Icon data={icon} color={getIconColor()} />}
+            <ItemText variant="cell_text" group="table" $active={active}>
+              {label}
+            </ItemText>
+            <Icon
+              data={showPanel ? chevron_up : chevron_down}
+              size={24}
+              color={typographyColor}
+            />
+          </AccordionButton>
+        </AccordionHeader>
+        {showPanel && (
+          <Panel
+            id={`panel_${accordionId}`}
+            role="region"
+            aria-labelledby={`header_${accordionId}`}
+          >
+            {children}
+          </Panel>
+        )}
+      </MenuItem>
+    )
+  }
+
+  const tooltip = menuIsOpen ? '' : label
+
+  return (
+    <>
+      <Tooltip title={tooltip} placement="right">
+        <MenuItem $active={active}>
+          {icon && (
+            <Button
+              ref={combinedRefs}
+              id="anchor-default"
+              aria-haspopup="true"
+              aria-expanded={menuIsOpen}
+              aria-controls="menu-default"
+              onClick={onClickWhenSidePanelClosed}
+              {...rest}
+            >
+              <Icon data={icon} color={getIconColor()} />
+              <AccordionIcon
+                data={arrow_drop_down}
+                size={24}
+                color={getIconColor()}
+              />
+            </Button>
+          )}
+        </MenuItem>
+      </Tooltip>
+      <Menu
+        open={menuIsOpen}
+        placement={'right-start'}
+        id="menu-default"
+        aria-labelledby="anchor-default"
+        onClose={closeMenu}
+        anchorEl={anchorEl}
+      >
+        {Children.map(children, (child) => {
+          const item = child as ReactElement<AccordionItemProps>
+          return (
+            <Menu.Item onClick={item.props.onClick}>
```

Comment:
Need to forward the `as` prop otherwise menu item is always a button when we want it to be an `a` or router `Link` etc